packages/medusa/src/services/order.ts

```diff
@@ -570,18 +638,55 @@ class OrderService extends TransactionBaseService {
         toCreate.no_notification = draft.no_notification_order
       }
 
-      const o = orderRepo.create(toCreate)
-      const result = await orderRepo.save(o)
+      const rawOrder = orderRepo.create(toCreate)
+      const order = await orderRepo.save(rawOrder)
 
       if (total !== 0 && payment) {
         await this.paymentProviderService_
           .withTransaction(manager)
           .updatePayment(payment.id, {
-            order_id: result.id,
+            order_id: order.id,
           })
       }
 
-      let gcBalance = await this.totalsService_.getGiftCardableAmount(cart)
+      if (!isDefined(cart.subtotal) || !isDefined(cart.discount_total)) {
+        throw new MedusaError(
+          MedusaError.Types.UNEXPECTED_STATE,
+          "Unable to compute gift cardable amount during order creation from cart. The cart is missing the subtotal and/or discount_total"
+        )
+      }
+
+      // gift cardable amount
+      /* const calculationContext = await this.totalsService_.getCalculationContext(
+         cart,
+         {
+           exclude_gift_cards: true
+         }
+       )
+       const totals = await this.newTotalsService_.getLineItemTotals(cart.items, {
+         includeTax: true,
+         calculationContext
+       })
+ 
+       let gcBalance = 0
+       if (cart.region?.gift_cards_taxable) {
+         const [subtotal, discountTotal] = Object.values(totals).reduce((acc, total) => {
+           acc[0] += total.total
+           acc[1] += total.discount_total
+           return acc
+         }, [0, 0])
+ 
+         gcBalance = subtotal - discountTotal
+       } else {
+         gcBalance = Object.values(totals).reduce((acc, total) => {
+           acc += total.total
+           return acc
+         }, 0)
+       }*/
```

Comment:
to remove?

Reply:
Nice catch 😅 it was a test of my own ahah